pkg/api/v1/routes/handlers.go

```diff
@@ -322,140 +299,124 @@ func (r *Routes) exclusiveNonFinalConditionExists(ctx context.Context, serverID
 	return nil
 }
 
-func (r *Routes) serverConditionDelete(c *gin.Context) {
+func (r *Routes) serverConditionDelete(c *gin.Context) (int, *v1types.ServerResponse) {
 	serverID, err := uuid.Parse(c.Param("uuid"))
 	if err != nil {
-		c.JSON(
-			http.StatusBadRequest,
-			&v1types.ServerResponse{Message: err.Error()},
-		)
 		r.logger.WithFields(logrus.Fields{
 			"serverID": c.Param("uuid"),
 		}).Info("bad serverID")
 
-		return
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
 	}
 
 	kind := ptypes.ConditionKind(c.Param("conditionKind"))
 	if !r.conditionKindValid(kind) {
-		c.JSON(
-			http.StatusBadRequest,
-			&v1types.ServerResponse{Message: "unsupported condition kind: " + string(kind)},
-		)
 		r.logger.WithFields(logrus.Fields{
 			"kind": kind,
 		}).Info("unsupported condition kind")
 
-		return
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "unsupported condition kind: " + string(kind),
+		}
 	}
 
 	// TODO (vc): the repository Delete route should take a condition id
```

Comment:
We need to discuss/ratify where and how the condition ID is to be used, its a bit blurry for me at the moment.

Reply:
In light of our conversation this morning, this code is fine as-is. I'll remove that comment.